### PR TITLE
Revise version language

### LINKED ISSUE #4414: Simplify language in the Release Channel section of Get Dart
### Page URL

https://dart.dev/get-dart#release-channels

### Page source

src/get-dart/index.md

### Describe the problem

The information within the **About release channels and version strings** section could use better organization. Recommend revising the content to better chunk the information into related blocks.

### Expected fix

Recommend using the following copy.

```markdown
### Stable channel
Dart deploys a release to the _stable_ channel about every three months.
The current stable version is  `[calculating]`{:.build-rev-stable}.

Use stable channel releases for production environments.

Versioning for releases in stable channel of the Dart SDK follows a `x.y.z` format.
The version strings resemble `1.24.3` and `2.1.0`.
This format consists of three dot-separated integers without hyphens or letters.
The versions include the major version (`x`), 
then the minor version (`y`), and then the patch version (`z`).

### Beta channel
Dart releases a build to the _preview_ channel about every month.
The current beta version is  `[calculating]`{:.build-rev-beta}.

These beta channel releases preview features intended for the stable channel.
Use releases in the beta channel to test new features or
test compatibility with future releases.
Don’t use beta releases for production environments. 

Versioning for releases in the **beta** channel of the Dart SDK
follows the `x.y.z-a.b.beta` format.
Version strings resemble `2.8.0-20.11.beta`.
The part before the hyphen follows the stable version format.
The part after the hyphen adds the pre-release version (`a`),
then the pre-release patch version (`b`), and then the channel (`beta`).

### Dev
Dart releases a build to the **dev** channel about twice a week.
The current dev version is  `[calculating]`{:.build-rev-dev}.

These dev channel builds include the most recent codebase with latest changes.
These builds may be broken, have no support,
and may contain unvetted breaking changes.

Versioning for releases in the **dev** channel of the Dart SDK
follows the `x.y.z-a.b.dev` format.
Version strings resemble `2.8.0-20.11.dev`.
Dev releases follow the beta channel versioning format replacing `beta` with `dev`.
```

### Additional context

Originally raised as part of https://github.com/dart-lang/site-www/pull/4410.

### PR DESCRIPTION
**Pull request by atsansone**
Fixes #4414

### CODE REVIEW

**Review by MaryaBelanger — COMMENTED** [review]
Thanks for doing this, I agree the version string format descriptions should just be coupled with the channel descriptions. 

Probably out of scope, but would each of these sections be a good place to include a quick:

> To switch to /whatever/ channel, run: `dart channel /whatever/"

? Idk if it actually works like that though (@johnpryan or DRE on rotation?)

File: src/get-dart/index.md [hunk 1]
```diff
@@ -78,33 +78,47 @@ Dart supports the following macOS versions as of November 2023:
...
+Version strings resemble `2.8.0-20.11.beta`.
+The part before the hyphen follows the stable version format.
+The part after the hyphen adds the pre-release version (`a`),
+then the pre-release patch version (`b`), and then the channel (`beta`).
+
+### Dev
+Dart releases a build to the **dev** channel about twice a week.
+The current dev version is  `[calculating]`{:.build-rev-dev}.
+
+These dev channel builds include the most recent codebase with latest changes.
```
[the remark this answers is not in the document]
Comment (atsansone) [reply src/get-dart/index.md:113]: **question:** Good question. Are we "releasing a build" or "deploying a release"? Probably the former, right?

File: src/get-dart/index.md [hunk 2]
```diff
@@ -78,33 +78,47 @@ Dart supports the following macOS versions as of November 2023:
...
-`1.24.3` and `2.1.0`. They consist of dot-separated integers, with no hyphens or
-letters, where `x` is the major version, `y` is the minor version, and `z` is
-the patch version.
-
-**Beta** and **dev** channel releases of the Dart SDK (non-stable releases) have
-`x.y.z-a.b.<beta|dev>` versions like `2.8.0-20.11.beta`. The part before the
-hyphen follows the stable version scheme, `a` and `b` after the hyphen are the
-prerelease and prerelease patch versions, and `beta` or `dev` is the channel.
+### Stable channel
+Dart deploys a release to the _stable_ channel about every three months.
```
[the remark this answers is not in the document]
Comment (johnpryan) [reply src/get-dart/index.md:82]: I would say something like "Dart publishes a new release to the stable channel about every three months". These are usually considered minor releases, major releases happen less often.

File: src/get-dart/index.md [hunk 3]
```diff
@@ -78,33 +78,47 @@ Dart supports the following macOS versions as of November 2023:
...
+The current stable version is  `[calculating]`{:.build-rev-stable}.
+
+Use stable channel releases for production environments.
+
+Versioning for releases in stable channel of the Dart SDK follows a `x.y.z` format.
```
Comment (johnpryan) [inline src/get-dart/index.md:87]: This paragraph is already covered elsewhere: https://dart.dev/tools/pub/versioning#semantic-versions

**Comment by atsansone** [comment]
Moved this change to https://github.com/dart-lang/site-www/pull/5497. Closing.

**Pull request closed without merging**